projects/vdk-plugins/vdk-ingest-http/tests/test_ingest_over_http.py

```diff
@@ -54,9 +58,62 @@ def test_ingest_over_http(mock_post):
     )
 
 
+@mock.patch("requests.post", side_effect=mocked_requests_post)
+def test_ingest_over_http_compression(mock_post):
+    threshold_bytes = 3
+    encoding = "utf-8"
+    job_context = MagicMock()
+    job_context.core_context.configuration.get_value.side_effect = [
```

Comment:
Why is side_effect being mocked and not return_value?

Reply:
Mocked return_value initially, however had issues. When mocking side_effect, values are properly set (verified in debug mode)